src/python/torchdistx/_C/fake.cc

```diff
@@ -6,15 +6,40 @@
 
 #include "module.h"
 
+#include <ATen/Context.h>
 #include <ATen/Tensor.h>
+#include <torch/csrc/utils/cuda_lazy_init.h>
 #include <torch/csrc/utils/pybind.h>
 #include <torchdistx/fake.h>
 
 namespace torchdistx::python {
+namespace {
+
+void pyEnterFakeMode(bool fake_cuda) {
+  enterFakeMode(fake_cuda);
+
+  // If CUDA is not available, suppress PyTorch's attempt to initialize its CUDA
+  // subsystem which would fail and prevent us from instantiating CUDA devices.
+  if (fake_cuda) {
+    if (!at::hasCUDA()) {
+      torch::utils::set_requires_cuda_init(false);
```

Comment:
is this a thread-local setting?

Reply:
It is global, but PyTorch's cuda-init implementation is already thread safe.